weave-js/src/components/PagePanelComponents/Home/Browse3/pages/MonitorsPage/CreateMonitorDrawer.tsx

```diff
@@ -0,0 +1,488 @@
+  onClose: () => void;
+  monitor?: ObjectVersionSchema;
+}) => {
+  const [error, setError] = useState<string | null>(null);
+  const [nameError, setNameError] = useState<string | null>(null);
+  const [description, setDescription] = useState<string>('');
+  const [monitorName, setMonitorName] = useState<string>('');
+  const [samplingRate, setSamplingRate] = useState<number>(10);
+  const [selectedOpVersionOption, setSelectedOpVersionOption] = useState<
+    string[]
+  >([]);
+  const [filter, setFilter] = useState<WFHighLevelCallFilter>({});
+  const [filterModel, setFilterModel] = useState<GridFilterModel>({items: []});
+  const {useCalls} = useWFHooks();
+  const [scorers, setScorers] = useState<{name: string; refUri?: string}[]>([]);
+  const [active, setActive] = useState<boolean>(false);
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+  const {sortBy, lowLevelFilter} = useFilterSortby(filter, {items: []}, [
+    {field: 'started_at', sort: 'desc'},
+  ]);
+  useMemo(() => {
+    const existingScorers = (monitor?.val['scorers'] || []).map(
+      (scorerRefUri: string) => ({
+        refUri: scorerRefUri,
+        name: parseRef(scorerRefUri).artifactName,
+      })
+    );
+    setMonitorName(monitor?.val['name'] || '');
+    setDescription(monitor?.val['description'] || '');
+    setSamplingRate((monitor?.val['sampling_rate'] || 0.1) * 100);
+    setFilter({opVersionRefs: monitor?.val['op_names'] || []});
+    setScorers(existingScorers);
+    setActive(monitor?.val['active'] || false);
+    setFilterModel(
+      queryToGridFilterModel(monitor?.val['query']) || {
+        items: [],
+      }
+    );
+  }, [monitor, setMonitorName]);
+
+  const {result: callsResults, loading: callsLoading} = useCalls(
+    entity,
+    project,
+    lowLevelFilter,
+    PAGE_SIZE,
+    PAGE_OFFSET,
+    sortBy
+  );
+
+  const tableData = useMemo(() => {
+    if (callsLoading || callsResults === null) {
+      return [];
+    }
+    return prepareFlattenedCallDataForTable(callsResults);
+  }, [callsResults, callsLoading]);
+
+  const {columns} = useCallsTableColumns(
+    entity,
+    project,
+    filter,
+    tableData,
+    new Set(),
+    () => {},
+    () => {},
+    () => false
+  );
+
+  const handleNameChange = useCallback(
+    (value: string) => {
+      setMonitorName(value);
+      const validationResult = validateDatasetName(value);
+      setNameError(validationResult.error);
+    },
+    [setNameError]
+  );
+
+  const opVersionOptions = useOpVersionOptions(entity, project, {});
+
+  useMemo(() => {
+    setSelectedOpVersionOption(filter.opVersionRefs ?? []);
+  }, [filter]);
+
+  const enableCreateButton = useMemo(() => {
+    return (
+      monitorName.length > 0 &&
+      selectedOpVersionOption.length > 0 &&
+      scorers.length > 0 &&
+      nameError === null
+    );
+  }, [monitorName, selectedOpVersionOption, scorers, nameError]);
+
+  const objCreate = useObjCreate();
+
+  const createScorer = useCallback(
```

Comment:
maybe i'll find out below - but this does not seem to do anything other than create a scorer of a class & name - no properties - maybe this is just for testing?

Reply:
At this time, this does not support parametrizing scorers from the UI. This is supported via the SDK though.

In order to support parametrizing scorers in the UI, we will need to do custom work for each scorer (custom form fields). So it would be good to define prioritization. I selected `ValidJSONScorer` as first to support since it doesn't require any parameters. I will add `ValidXMLScorer` too.
Then we can start `LLMAsAJudgeScorer` for parametrizable scorers.